clickhouse-admin/src/clickhouse_cli.rs

```diff
@@ -102,6 +105,22 @@ impl ClickhouseCli {
         .await
     }
 
+    pub async fn keeper_cluster_membership(
+        &self,
+    ) -> Result<ClickhouseKeeperClusterMembership, ClickhouseCliError> {
+        let lgif_output = self.lgif().await?;
+        let conf_output = self.keeper_conf().await?;
+        let raft_output = self.raft_config().await?;
+        let raft_config: BTreeSet<KeeperId> =
+            raft_output.keeper_servers.iter().map(|s| s.server_id).collect();
+
+        Ok(ClickhouseKeeperClusterMembership {
+            queried_keeper: conf_output.server_id,
```

Comment:
This works fine, but I think it will be unnecessary in the near future.

What I was talking about is that when we write the keeper and server configs, we should also store an extra file that contains the generation number from the `KeeperConfigurableSettings` and `ServerConfigurableSettings` of the last written config. We would cache that generation number in the `clickhouse-admin` server and load it from the file on startup. We need to do this as an optimization so that we don't constantly overwrite the clickhouse config files and force them to be reloaded when nothing has changed.

I was thinking that instead of just saving the generation number we could save the entire `ServerConfigurableSettings` and `KeeperConfigurableSettings` and cache those as well. Doing that would allow us to just look in memory of the `clickhouse-admin` server for the keeper-id. And since we use these settings to write the keeper config files, the values would always be the same.